crm-project/src/main/java/CRM/controller/facades/SettingsFacade.java

```diff
@@ -15,10 +26,27 @@ public class SettingsFacade {
     @Autowired
     private SettingsService settingsService;
 
+    /**
+     * This method is used to retrieve user's notification settings in a specified board.
+     *
+     * @param userId The id of the user whose settings are to be retrieved.
+     * @param boardID The id of the board whose user's settings belong to.
+     * @return A Response object containing all the retrieved settings or an error message if the user/board is not found or the id is invalid.
+     * @throws IllegalArgumentException if the specified user/board id is invalid.
+     * @throws NullPointerException     if the specified user/board id is null.
+     * @throws NoSuchElementException   if the user with the specified id is not found/user does not belong to that board.
+     */
     public Response getAllUserSettingsInBoard(Long userId, Long boardID){
-        // validate the parameters using Validations.validate function
-        // call settingsService and ask for all the settings for this user in a specific board
-        return null;
+        try {
+            Validations.validate(userId, Regex.ID.getRegex());
+            Validations.validate(boardID, Regex.ID.getRegex());
```

Comment:
We have a function that checks all the ids inside Validations class
l think it's called Validations.validateIDs or something like that. Look at it.

Reply:
fixed